OfflineAudioContext/explainer.md

```diff
@@ -0,0 +1,232 @@
+
+Another implication is that the audio buffer cannot be easily interleaved with video data streamed out of WebCodecs. To use clipchamp.com again as an example, the video and audio are combined into a .mp4 file during the export process. The video and audio streams need to be interleaved/muxed in the correct order before writing to the file; the audio data cannot simply be appended at the end. Ignoring the memory implications of the current API, it is difficult to interleave video and audio when all the audio data is delivered as a single chunk at the end of processing. If the audio data was streamed out at the same time as video data is streamed out of WebCodecs then it would simplify the interleaving process.
+
+A workaround to these limitations is for developers to build custom WASM audio-processing which can stream out data incrementally so that the full AudioBuffer is not allocated and therefore memory pressure is not applied to a user's machine. While this works around the API constraint, these 3rd party libraries require complex integration, increase maintenance burden for developers and can introduce variability across platforms which can lead to poor use experiences. Custom WASM libraries duplicate features that already exist in WebAudio and only provide streaming output support as a benefit.
+
+### Goals
+
+- Allow streaming data out of a WebAudio in an offline context for rendering large audio graphs
+
+### Non-goals
+
+- Change the existing `startRendering()` behavior, this API change is additive
+
+## Proposed Approach - Add `startRenderingStream()` function
+
+The preferred approach is adding a new method `startRenderingStream()` that yields buffers of interleaved audio samples in a Float32Array, or another format as outlined in Open Questions. In this scenario, the user can read chunks as they arrive and consume them for storage, transcoding via WebCodecs, sending to a server, etc.
+
+```js
+// From https://developer.mozilla.org/en-US/docs/Web/API/AudioData/format
+enum AudioFormat {
+    "u8",
+    "s16",
+    "s32",
+    "f32",
+    "u8-planar",
+    "s16-planar",
+    "s32-planar",
+    "f32-planar"
+}
+
+dictionary OfflineAudioRenderingOptions {
+    required AudioFormat format = "f32";
+}
+
+partial interface OfflineAudioContext {
+    // Returns a stream that yields buffers of interleaved audio samples in Float32Array or whatever format is specified
+    Promise<ReadableStream> startRenderingStream(options?: OfflineAudioRenderingOptions);
+};
+```
+
+Usage example:
+
+```js
+const context = new OfflineAudioContext(2, 44100, 44100);
+
+// Add some nodes to build a graph...
+
+const reader = context.startRenderingStream({ format: "f32-planar" }).reader();
+while (true) {
```

Comment:
nit: Let's avoid `while (true)` in our example.  We can just make up some terminal state instead:

```
while (!isRenderingCompleted) {
...
}
```